app/components/QrCodeScanner/QrCodeScanner.jsx

```diff
@@ -29,24 +51,99 @@ export default class QrCodeScanner extends Component<Props> {
       callback(content, this.stopScanner.bind(this))
     })
 
-    Instascan.Camera.getCameras()
+    // since the browser halts while getting usermedia
+    // let the loading animation finish one round and only then continue
+    this.setState({ loading: true })
+    new Promise(resolve => setTimeout(resolve, 900))
+      .then(() => Instascan.Camera.getCameras())
       .then((cameras: Array<Object>) => {
-        if (cameras.length > 0) {
-          this.scannerInstance.start(cameras[0])
-        } else {
-          console.error('No cameras found.')
+        if (cameras.length === 0) {
+          // shouldn't happen, case covered by withCameraAvailability
+          throw new Error('No cameras found.')
         }
+        return this.scannerInstance.start(cameras[0])
+      })
+      .catch(err => {
+        this.setState({ error: this.constructor.getScannerError(err) })
+      })
+      .finally(() => {
+        this.setState({ loading: false })
       })
-      .catch(e => console.error(e))
   }
 
   stopScanner() {
     if (this.scannerInstance) this.scannerInstance.stop()
+    this.setState({ error: null }) // clear error
+  }
+
+  static getScannerError(err: Error) {
+    const scanErr: ScannerError = {
+      message: 'Could not connect to camera'
+    }
+
+    if (err.name === 'TrackStartError') {
+      // get link info by user os, defaults to nothing
+      const [link, title] =
+        {
+          darwin: [
+            'https://support.apple.com/en-il/guide/mac-help/mh32356/10.14/mac',
+            'MacOS User Guide: Change Privacy preferences'
+          ],
+          win32: [
+            'https://support.microsoft.com/en-ca/help/10557/windows-10-app-permissions',
+            'Windows Support: App permissions'
+          ],
+          linux: [
+            'https://wiki.ubuntu.com/SecurityPermissions',
+            'Ubuntu Wiki: Security Permissions'
+          ]
+        }[process.platform] || []
```

Comment:
where is process defined? perhaps this is a good use case for lodash get

Reply:
`process` is an Electron global variable.

I'll go for lodash.get 👍🏻